core/src/main/java/com/linecorp/armeria/server/annotation/BeanRequestConverterFunction.java

```diff
@@ -0,0 +1,320 @@
+
+        private void setValue(final ServiceRequestContext ctx,
+                              final AggregatedHttpMessage request,
+                              final MutableValueHolder<HttpParameters> httpParams,
+                              final Object targetBean) {
+            final String strValue = paramValueRetriever.getParamValue(ctx, request, httpParams);
+
+            if (strValue == null) {
+                return;
+            }
+
+            final Object paramValue = convertParameter(strValue, type, enumConverter, false);
+
+            setParamValue(targetBean, paramValue);
+        }
+    }
+
+    private abstract static class FieldParamInfo extends ParamInfo {
+        private final Field field;
+
+        private FieldParamInfo(final Field field,
+                               final ParamValueRetriever paramValueRetriever) {
+            super(field.getType(), paramValueRetriever);
+            this.field = field;
+            this.field.setAccessible(true);
+        }
+
+        @Override
+        protected void setParamValue(final Object targetBean,
+                                     final Object paramValue) {
+            try {
+                field.set(targetBean, paramValue);
+            } catch (final Exception e) {
+                throw new IllegalArgumentException(
+                        "Can't set '" + paramValue + "' " + "to field '" +
+                        field.getDeclaringClass().getCanonicalName() + '.' + field.getName() + "'.",
+                        e);
+            }
+        }
+    }
+
+    /**
+     * a field annotated by {@link Param}.
+     */
+    private static final class ParamFieldParamInfo extends FieldParamInfo {
+        private ParamFieldParamInfo(final Field field, final Param param) {
+            super(field, ParamValueRetriever.fromParam(param.value()));
+        }
+    }
+
+    /**
+     * a field annotated by {@link Header}.
+     */
+    private static final class HeaderFieldParamInfo extends FieldParamInfo {
+        private HeaderFieldParamInfo(final Field field, final Header header) {
+            super(field, ParamValueRetriever.fromHeader(header.value()));
+        }
+    }
+
+    private abstract static class MethodParamInfo extends ParamInfo {
+        private final Method method;
+
+        private MethodParamInfo(final Method method, final ParamValueRetriever paramValueRetriever) {
+            super(method.getParameterTypes()[0], paramValueRetriever);
+            this.method = method;
+            this.method.setAccessible(true);
+        }
+
+        @Override
+        protected void setParamValue(final Object targetBean,
+                                     final Object paramValue) {
+            try {
+                method.invoke(targetBean, paramValue);
+            } catch (final Exception e) {
+                throw new IllegalArgumentException(
+                        "Can't use '" + paramValue + "' " + " to invoke method '" +
```

Comment:
`paramValue + "' " + " to invoke method '"` -> `paramValue + "' to invoke method '"`

Reply:
these code was changed to `String.format('message template', msgParams)`